solver/src/main/java/org/chocosolver/solver/constraints/nary/binPacking/PropBinPacking.java

```diff
@@ -133,10 +133,6 @@ public PropBinPacking(IntVar[] itemBin, int[] itemSize, IntVar[] binLoad, int of
                 if(itemBin[i].contains(j+offset)) {
```

Comment:
shouldn't this whole block (setting of P and R) be inside the fullPropagation method (last method) ?

Reply:
It could, and it might be more consistent in terms of code. In terms of correctness, I think it does not change anything so  either way (let the code inside the constructor or at the beginning of the fullPropagation part) is ok for me

Reply:
Yes, it is mostly for readability. 
I remember though having encountered an issue with @cprudhom many years ago because of setting up data structure in the constructor, in a specific context where a fullPropagation was called during search by the engine. Unfortunately, I do not remember the details and maybe the involved sets where not stored, in which case we understand easily why it didn't work at that time. @cprudhom does it ring a bell ?

Reply:
I move the code, initialising P and sumP, from the constructor to the beginning of the fullPropagation.

Reply:
> @cprudhom does it ring a bell ?

As soon as there are data structures, you have to be careful about initializing them, that's true. As the variables may have been modified since the call to the constructor, the structures may not be up to date, which is why it is always better to delay populating them until the initial propagation.